setup.py

```diff
@@ -256,7 +256,7 @@ def run_tests(self):
     ],
     packages=find_packages(exclude=(TESTS_DIRECTORY,)),
     install_requires=[
-        'sqlalchemy>=1.0.0',
+        'sqlalchemy>=2.0.0',
```

Comment:
There is a lot of changes in SQLAlchemy since v2.0. I understand your point but I've the feeling that's easier to maintain two branches if necessary than maintain all the complexity in the source code.